src/lsm/lsm_worker.c

```diff
@@ -93,11 +92,13 @@ __lsm_worker(void *arg)
 
 	cookie = (WT_LSM_WORKER_ARGS *)arg;
 	session = cookie->session;
-	conn = S2C(session);
 
 	entry = NULL;
-	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
-	    F_ISSET(cookie, WT_LSM_WORKER_RUN)) {
+	for (;;) {
+		WT_BARRIER();			/* Don't cache the run flag. */
```

Comment:
@agorrod 
> I'd much rather not have this WT_BARRIER here - it's going to make me think every time I encounter the code.

I've changed this to using a new `WT_LSM_WORKER_ARGS.running` field that's declared volatile, in commit e65259f.